File: app/src/main/java/org/jeonfeel/moeuibit2/MyCoin.java
package org.jeonfeel.moeuibit2;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class MyCoin {

    @PrimaryKey
    @NonNull
    public String market;
    public Double purchasePrice;
    public String koreanCoinName;
    public String symbol;
    public Double quantity;

    public MyCoin(@NonNull String market, Double purchasePrice, String koreanCoinName, String symbol, Double quantity) {
        this.market = market;
        this.purchasePrice = purchasePrice;
        this.koreanCoinName = koreanCoinName;
        this.symbol = symbol;
        this.quantity = quantity;
    }

    @NonNull
    public String getMarket() {
        return market;
    }

    public void setMarket(@NonNull String market) {
        this.market = market;
    }

    public Double getPurchasePrice() {
        return purchasePrice;
    }

    public void setPurchasePrice(Double purchasePrice) {
        this.purchasePrice = purchasePrice;
    }

    public String getKoreanCoinName() {
        return koreanCoinName;
    }

    public void setKoreanCoinName(String koreanCoinName) {
        this.koreanCoinName = koreanCoinName;
    }

    public String getSymbol() {
        return symbol;
    }

    public void setSymbol(String symbol) {
        this.symbol = symbol;
    }

    public Double getQuantity() {
        return quantity;
    }

    public void setQuantity(Double quantity) {
        this.quantity = quantity;
    }

}

File: app/src/main/java/org/jeonfeel/moeuibit2/data/local/room/entity/MyCoin.kt
package org.jeonfeel.moeuibit2.data.local.room.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import org.jeonfeel.moeuibit2.ui.main.portfolio.dto.UserHoldCoinDTO

@Entity(primaryKeys = ["market", "exchange"])
data class MyCoin(
    var market: String = "",
    var exchange: String = "",
    var purchasePrice: Double = 0.0,
    var koreanCoinName: String = "",
    var symbol: String = "",
    var quantity: Double = 0.0,
    @ColumnInfo(defaultValue = "0.0") var purchaseAverageBtcPrice: Double = 0.0
) {
    fun parseUserHoldsModel(): UserHoldCoinDTO {
        return UserHoldCoinDTO(
            myCoinKoreanName = "",
            myCoinEngName = "",
            myCoinsSymbol = this.symbol,
            myCoinsQuantity = quantity,
            myCoinsBuyingAverage = purchasePrice,
            currentPrice = 0.0,
            openingPrice = 0.0,
            warning = false,
            isFavorite = 0,
            caution = null,
            market = this.market,
            purchaseAverageBtcPrice = this.purchaseAverageBtcPrice
        )
    }

    fun isEmpty(): Boolean {
        return this.quantity == 0.0
                || this.purchasePrice == 0.0
                || this.market == ""
                || this.symbol == ""
    }
}